components/support/utils/src/test/java/mozilla/components/support/utils/URLStringUtilsTest.kt

```diff
@@ -67,9 +67,21 @@ class URLStringUtilsTest {
     @Test
     fun isSearchTerm() {
         assertTrue(isSearchTerm("inurl:mozilla.org advanced search"))
-        assertTrue(isSearchTerm("3.14.2019"))
+        assertTrue(isSearchTerm("sf: help"))
+        assertTrue(isSearchTerm("mozilla./~"))
+        assertTrue(isSearchTerm("cnn.com politics"))
 
         assertFalse(isSearchTerm("about:config"))
+        assertFalse(isSearchTerm("about:config:8000"))
         assertFalse(isSearchTerm("http://192.168.255.255"))
+        // Per https://bugs.chromium.org/p/chromium/issues/detail?id=31405, ICANN will accept
+        // purely numeric gTLDs.
+        assertFalse(isSearchTerm("3.14.2019"))
+        assertFalse(isSearchTerm(" cnn.com "))
+        assertFalse(isSearchTerm(" cnn.com"))
+        assertFalse(isSearchTerm("cnn.com "))
+        assertFalse(isSearchTerm("mozilla.com/~userdir"))
+        assertFalse(isSearchTerm("http://faß.de//"))
```

Comment:
I noticed our algorithm looks for one of `://`, `:`, and `.`: that means `http://faß.de//` is a significantly different test case than, `faß.de`. Should we include both?

Maybe this is why the tests didn't catch the non-unicode error before (if I understood you correctly).

Reply:
Yes. I think.

I believe that what happened was the unicode B matched \s* in that case because it fell after the ://. When it was before the . (the first required punctuation element) (as in the latter example), it would not match in the `\w` until I passed the proper option to the `compile` method. I see no reason to change our approach, but I suppose I haven't thought about it too deeply and reserve my ability to change my mind :-)

Reply:
I do think it a bit odd that the protocol and TLD are handled in the same branch but, given the guided principle of,

> Be lenient about what is classified as potentially a URL. Anything that contains ://, or . and has no internal spaces is potentially a URL.

Maybe it's good enough? In any case, I think landing this as is – as a glorified back-out to improve perf quickly – and handing it off to the a-c if they feel the implementation needs to be more robust is a fine approach for us.

Reply:
This is definitely a very lenient approach to deciding what is a URL. This is exactly the method that Fennec uses, so I think it is reasonable.